venom/juce/juce_gui_basics/Slider/Slider.cpp

```diff
@@ -1,34 +1,42 @@
 #include "Slider.h"
 #include <juce_gui_basics/juce_gui_basics.h>
 
-void init_Slider(py::module& m) {
-        auto slider = py::class_<juce::Slider, juce::Component>(m, "Slider", py::dynamic_attr())
-        .def(py::init<>())
-        .def(py::init([](const juce::String& componentName)
-                { return new juce::Slider(componentName); }))
-        .def(py::init([](juce::Slider::SliderStyle style, juce::Slider::TextEntryBoxPosition textBoxPosition)
-                { return new juce::Slider(style, textBoxPosition); }))
-        .def("setRange", [](juce::Slider &self, double newMinimum, double newMaximum, double newInterval)
-            { self.setRange(newMinimum, newMaximum, newInterval); })
-        .def("getValue", [](juce::Slider &self)
-            { return self.getValue(); })
-        .def("setSliderStyle", [](juce::Slider &self, juce::Slider::SliderStyle newStyle)
-            { self.setSliderStyle(newStyle); })
-        .def("getSliderStyle", [](juce::Slider &self)
-            { return self.getSliderStyle(); })
-        .def("setTextBoxStyle", [](juce::Slider &self, juce::Slider::TextEntryBoxPosition newPosition,
-                          bool isReadOnly,
-                          int textEntryBoxWidth,
-                          int textEntryBoxHeight)
-            { self.setTextBoxStyle(newPosition, isReadOnly, textEntryBoxWidth, textEntryBoxHeight); })
-        .def("setVisible", [](juce::Slider &self, bool shouldBeVisible)
-            { self.setVisible(shouldBeVisible); })
-        .def("setBounds", [](juce::Slider &self, int x, int y, int w, int h)
-            {  self.setBounds(x, y, w, h); })
-        .def("setSize", [](juce::Slider &self, int newWidth, int newHeight)
-            {  self.setSize(newWidth, newHeight); });
-        
-        py::enum_<juce::Slider::SliderStyle>(slider, "SliderStyle")
+void init_Slider(py::module &m) {
+//    py::class_<juce::Component>(m, "Component").def(py::init<>());
+    auto slider = py::class_<juce::Slider, juce::Component>(m, "Slider", py::dynamic_attr())
+            .def(py::init<>([]() {
+                return createOnMessageThread<juce::Slider>([] {
```

Comment:
Maybe we could just implement it in the parent class. Or maybe there is a way to make this initialization uniform for all UI components without the need of explicitly doing it every time